models/collection.go

```diff
@@ -215,128 +201,15 @@ func (c *CollectionApi) body() (body []byte, err error) {
-			attempt, statusCode, http.StatusText(statusCode),
-			time.Since(start).String()))
-		if err != nil {
-			if !c.shouldRetry(method, statusCode, err) {
-				tflog.Error(ctx, fmt.Sprintf("%v %v (%v) failed: %v",
-					attempt, method, url, statusCode))
-				responseBody = nil
-				return
-			}
+func (c *CollectionApi) handleRequestError(ctx context.Context, method string, statusCode int, responseBody []byte, requestErr error) (shouldRetry bool, newStatusCode int, newBody []byte, err error) {
+	newStatusCode, newBody, err = statusCode, responseBody, requestErr
 
-			if ctx.Err() == nil {
-				attempt++
-				continue
-			}
-		}
-		break
-	}
-
-	if err == nil {
-		err = ctx.Err()
-	}
-
-	if err != nil {
-		tflog.Error(ctx, fmt.Sprintf("%v %v (%v) failed: %s", method, url,
-			attempt, err.Error()))
-	}
-	return
-}
-
-func (c *CollectionApi) request(ctx context.Context, method string, u string, body []byte) (statusCode int, responseBody []byte, err error) {
-	client := clients.Get(c.splunk)
-	req, err := http.NewRequestWithContext(ctx, method, u, bytes.NewBuffer(body))
-	if err != nil {
-		return
-	}
-	if Verbose {
-		err = logRequest(req)
-		if err != nil {
-			return
-		}
-	}
-
-	if c.splunk.BearerToken != "" {
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.splunk.BearerToken))
-	} else {
-		req.SetBasicAuth(c.splunk.User, c.splunk.Password)
-	}
-	if c.apiConfig.ContentType != "" {
-		req.Header.Add("Content-Type", c.apiConfig.ContentType)
-	} else if c.apiConfig.BodyFormat == "JSON" {
-		req.Header.Add("Content-Type", "application/json")
-	}
-
-	itsiLimiter.Acquire()
-	defer itsiLimiter.Release()
-
-	tflog.Debug(ctx, "COLLECTION: Created a request",
-		map[string]interface{}{"key": c.RESTKey, "method": method, "url": u})
-
-	tflog.Trace(ctx, "COLLECTION:   Request body",
-		map[string]interface{}{"key": c.RESTKey, "c": c, "body": string(body)})
-
-	resp, err := client.Do(req)
-	if resp != nil {
-		statusCode = resp.StatusCode
-	}
-	if err != nil {
-		tflog.Trace(ctx, "COLLECTION:   Request fail", map[string]interface{}{"key": c.RESTKey, "err": err})
-		return
-	}
-	defer resp.Body.Close()
-	tflog.Debug(ctx, "COLLECTION:   Response err code: "+resp.Status,
-		map[string]interface{}{"key": c.RESTKey, "content-length": resp.ContentLength})
-	if Verbose {
-		if err = logResponse(resp); err != nil {
-			return
-		}
-	}
-
-	responseBody, err = io.ReadAll(resp.Body)
-	if err != nil {
-		tflog.Trace(ctx, "COLLECTION:   Response read err",
-			map[string]interface{}{"key": c.RESTKey, "err": err})
-		return statusCode, nil, fmt.Errorf("%v error: %v", method, resp.Status)
-	}
-
-	success := false
-	switch method {
-	case http.MethodDelete:
-		if resp.StatusCode == 404 {
-			// Ignore 404 errors for DELETE requests, when
-			// we are trying to delete a nonexistent
-			// resource...
-			success = true
-			break
-		}
-		success = resp.StatusCode >= 200 && resp.StatusCode < 300
+	switch {
```

Comment:
the same codes from the removed map:
https://github.com/TiVo/terraform-provider-splunk-itsi/pull/540/files#diff-b91cfe5ce6144942909a8ff1ed350a22e385907ec19f37eaa613d938f5e17b64L165-L176

Decided to keep this function as separate implementation of base case it has 'extra logic' to handle conflicts & delete.